File: Windows/PS Classics fPKG Builder/PS2Game.vb
Imports System.IO

Public Class PS2Game

    Public Shared Function GetPS2GameID(GameISO As String) As String
        Dim GameID As String = ""

        Using SevenZip As New Process()
            SevenZip.StartInfo.FileName = My.Computer.FileSystem.CurrentDirectory + "\Tools\7z.exe"
            SevenZip.StartInfo.Arguments = "l -ba """ + GameISO + """"
            SevenZip.StartInfo.RedirectStandardOutput = True
            SevenZip.StartInfo.UseShellExecute = False
            SevenZip.StartInfo.CreateNoWindow = True
            SevenZip.Start()

            'Read the output
            Dim OutputReader As StreamReader = SevenZip.StandardOutput
            Dim ProcessOutput As String() = OutputReader.ReadToEnd().Split(New String() {vbCrLf}, StringSplitOptions.None)

            If ProcessOutput.Length > 0 Then
                For Each Line As String In ProcessOutput
                    If Line.Contains("SLES_") Or Line.Contains("SLUS_") Or Line.Contains("SCES_") Or Line.Contains("SCUS_") Then
                        If Line.Contains("Volume:") Then 'ID found in the ISO Header
                            If Line.Split(New String() {"Volume: "}, StringSplitOptions.RemoveEmptyEntries).Length > 0 Then
                                GameID = Line.Split(New String() {"Volume: "}, StringSplitOptions.RemoveEmptyEntries)(1)
                                Exit For
                            End If
                        Else 'ID found in the ISO files
                            If String.Join(" ", Line.Split(New Char() {}, StringSplitOptions.RemoveEmptyEntries)).Split(" "c).Length > 4 Then
                                GameID = String.Join(" ", Line.Split(New Char() {}, StringSplitOptions.RemoveEmptyEntries)).Split(" "c)(5).Trim()
                                Exit For
                            End If
                        End If
                    End If
                Next
            End If

        End Using

        Return GameID
    End Function

    Public Shared Function GetPS2GameTitleFromDatabaseList(GameID As String) As String
        Dim FoundGameTitle As String = ""
        GameID = GameID.Replace("-", "")

        For Each GameTitle As String In File.ReadLines(My.Computer.FileSystem.CurrentDirectory + "\Tools\ps2ids.txt")
            If GameTitle.Contains(GameID) Then
                FoundGameTitle = GameTitle.Split(";"c)(1)
                Exit For
            End If
        Next

        If String.IsNullOrEmpty(FoundGameTitle) Then
            Return "Unknown PS2 game"
        Else
            Return FoundGameTitle
        End If
    End Function

End Class

File: Linux/makePSClassicsfPKG/PS2Game.cs
using System;
using System.Diagnostics;
using System.IO;

namespace makePSClassicsfPKG
{

    public class PS2Game
    {

        public static string GetPS2GameID(string GameISO)
        {
            string GameID = "";
            using var Bash = new Process();

            var StringsCMD = $"7z l -ba \"{GameISO}\"";
            var EscapedArgs = StringsCMD.Replace("\"", "\\\"");

            Bash.StartInfo.FileName = "/bin/bash";
            Bash.StartInfo.Arguments = $"-c \"{EscapedArgs}\"";
            Bash.StartInfo.RedirectStandardOutput = true;
            Bash.StartInfo.RedirectStandardError = true;
            Bash.StartInfo.UseShellExecute = false;
            Bash.StartInfo.CreateNoWindow = false;
            Bash.Start();
            Bash.WaitForExit();

            var OutputReader = Bash.StandardOutput;
            string[] ProcessOutput = OutputReader.ReadToEnd().Split(new string[] { "\r\n" }, StringSplitOptions.RemoveEmptyEntries);

            if (ProcessOutput.Length > 0)
            {
                foreach (string Line in ProcessOutput)
                {
                    if (Line.Contains("SLES_") | Line.Contains("SLUS_") | Line.Contains("SCES_") | Line.Contains("SCUS_"))
                    {
                        if (Line.Contains("Volume:")) // ID found in the ISO Header
                        {
                            if (Line.Split(new string[] { "Volume: " }, StringSplitOptions.RemoveEmptyEntries).Length > 0)
                            {
                                GameID = Line.Split(new string[] { "Volume: " }, StringSplitOptions.RemoveEmptyEntries)[1];
                                break;
                            }
                        }
                        else if (string.Join(" ", Line.Split(new char[] { }, StringSplitOptions.RemoveEmptyEntries)).Split(' ').Length > 4) // ID found in the ISO files
                        {
                            GameID = string.Join(" ", Line.Split(new char[] { }, StringSplitOptions.RemoveEmptyEntries)).Split(' ')[5].Trim();
                            break;
                        }
                    }
                }
            }

            return GameID;
        }

        public static string GetPS2GameTitleFromDatabaseList(string GameID)
        {
            string FoundGameTitle = "";
            GameID = GameID.Replace("-", "");

            foreach (string GameTitle in File.ReadLines(Directory.GetCurrentDirectory() + @"/Tools/ps2ids.txt"))
            {
                if (GameTitle.Contains(GameID))
                {
                    FoundGameTitle = GameTitle.Split(';')[1];
                    break;
                }
            }

            if (string.IsNullOrEmpty(FoundGameTitle))
            {
                return "Unknown PS2 game";
            }
            else
            {
                return FoundGameTitle;
            }
        }

    }
}